### PR TITLE
feat(core): add stage status precondition type

### PR DESCRIPTION
**Pull request by maggieneterval**
UI for https://github.com/spinnaker/spinnaker/issues/4768 (Orca PR coming soon)
@louisjimenez I was kinda surprised that this took editing so many files to make work so let me know if you notice that I'm doing something dumb, I don't really remember how to use angular.

beb7f081-3c46-4c35-b007-7d4887aa46c0

### CODE REVIEW

**Comment by maggieneterval** [comment]
@ethanfrogers I know I'm so sorry...the "Check" select is Angular and looks aligned with the other Angular-y options (expression, cluster size) but I neither wanted to 1) add more Angular or 2) rewrite the other options to use FormikField / React :cry: :cry:

File: app/scripts/modules/core/src/pipeline/config/preconditions/types/stageStatus/StageStatusPreconditionConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  updatePreconditionContext,
+}: IStageStatusPreconditionConfigProps) {
+  return (
+    <div className="form-group row">
+      <FormField
+        input={inputProps => (
+          <ReactSelectInput
+            {...inputProps}
+            clearable={false}
+            options={upstreamStages.map(stage => ({ value: stage.name, label: stage.name }))}
```
Comment (louisjimenez) [inline app/scripts/modules/core/src/pipeline/config/preconditions/types/stageStatus/StageStatusPreconditionConfig.tsx]: _supermicro_ nit: could be destructed  `map(({ name }) => ({ value: name, label: name }))`

**Comment by maggieneterval** [comment]
Alright after playing around with this for a bit and encountering more complex workflows than succeed/fail (most notably, fail/continue) I decided to just make every possible execution status available so that any status-checking SpEL can be replaced by this feature.

**Comment by maggieneterval** [comment]
(Will merge once corresponding Orca PR is merged: https://github.com/spinnaker/orca/pull/3116)
:heart: :heart: :heart:

**Pull request merged**